src/Neo/Cryptography/BN254.cs

```diff
@@ -0,0 +1,277 @@
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// BN254.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Nethermind.MclBindings;
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Neo.Cryptography
+{
+    public static class BN254
+    {
+        public const int FieldElementLength = 32;
+        public const int G1EncodedLength = 64;
+        public const int PairInputLength = 192;
+
+        private static readonly object s_sync = new();
+        private static bool s_initialized;
```

Comment:
This is not thread safe. The `s_initialized` must be volatile.

```suggestion
        private static volatile bool s_initialized;
```